base/errorshow.jl

```diff
@@ -248,9 +248,9 @@ function showerror(io::IO, ex::MethodError)
     if f === Base.convert && length(arg_types_param) == 2 && !is_arg_types
         f_is_function = true
         show_convert_error(io, ex, arg_types_param)
-    elseif isempty(methods(f)) && isa(f, DataType) && f.abstract
+    elseif (allmethods = methods(f); isempty(allmethods)) && isa(f, DataType) && f.abstract
```

Comment:
```suggestion
    elseif isa(f, DataType) && f.abstract && isempty(methods(f))
```

Reply:
I've got to say though, this conditional makes absolutely no sense to me. I think I get what it's trying to test, but just that it appears completely useless in its current form
```suggestion
    elseif isa(f, Type) && try
        isempty(_methods_by_ftype(Tuple{Type{<:Base.typename(f).wrapper}, Vararg}, -1, typemax(UInt64)));
        catch ex; ex isa ErrorException ? false : rethrow(); end # try/catch because typename may throw for Union
```

Reply:
Is it still worthwhile to try to fix, or simply delete, this code branch?